minor changes to config files

ctf0 opened the pull request:
1- changed some security settings to make it more secure out-of-the-box
2- changed the data-upload limit for something more reasonable
3- added an env key for `SESSION_DOMAIN` for easier user exp
4- fixed some docs typos and made config files more consistent
5- added `PATCH` verb to **csrf**

ctf0 committed:
cleanup

1- changed some security settings to make it as more secure out of the
box
2- changed the data-upload limit for something more reasonable
3- added an env key for SESSION_DOMAIN for easier user exp
4- fixed some docs typos and made config files more consistent
5- added PATCH verb to csrf

ctf0 committed:
added the new session key to env

RomainLanz commented:
Nice one !

ctf0 commented:
i will update the docs once the PR is merged.

thetutlage commented on config/bodyParser.js:
```diff
@@ -1,24 +1,26 @@
   |
   */
-  limit: '1mb',
+  limit: '125mb',
```
125MB for post data is too much. This does not include file uploads and should be under `20MB`. If someone has requirements they should come and toggle this value, atleast with that they would know what they are doing.

ctf0 replied:
okay no problem

ctf0 committed:
more cleanup

1- add all the supported db config
2- change default db
3- now the error msg in the view make sense
4- order ace commands alpha

RomainLanz commented on .env.example:
```diff
@@ -4,8 +4,10 @@ APP_KEY=n96M1TPG821EdN4mMIjnGKxGytx9W2UJ
 SESSION_DRIVER=cookie
-DB_CONNECTION=sqlite
+SESSION_DOMAIN=example.com
+DB_CONNECTION=mysql
```
We should keep sqlite

ctf0 replied:
the reason i changed because most of ppl use mysql and to also keep the db env keys unless the user wants to use sqlite then he can remove them

RomainLanz replied:
SQLite driver can works without installing anything (only the sqlite package). That's why it is use by default.

ctf0 replied:
no problem, we can change it back then, but in that case the other keys for db in env is useless, what if i removed it aswell and in case user needs a different driver then he can add what he wants ?

RomainLanz commented on .env.example:
```diff
@@ -4,8 +4,10 @@ APP_KEY=n96M1TPG821EdN4mMIjnGKxGytx9W2UJ
+SESSION_DOMAIN=example.com
+DB_CONNECTION=mysql
 DB_HOST=127.0.0.1
+DB_PORT=3306
```
This is the default port

ctf0 replied:
yes but sometimes the user use a different one, so its kept just incase

RomainLanz commented on config/database.js:
```diff
@@ -18,13 +18,14 @@ module.exports = {
   /*
   |--------------------------------------------------------------------------
-  | Sqlite
+  | Drivers
```
Why?

ctf0 replied:
because from the docs, some dbs have more than one driver, so its left for the user to decide which driver he wants to install/use

RomainLanz commented on config/database.js:
```diff
@@ -35,43 +36,64 @@ module.exports = {
-  |
-  | npm i --save mysql
-  |
-  */
```
Why?

RomainLanz commented on config/database.js:
```diff
@@ -35,43 +36,64 @@ module.exports = {
+    }
+  },
+
+  maria: {
```
It's the same as mysql

ctf0 replied:
i know, but i added as it was mentioned in the docs, so i thought maybe there is something different

RomainLanz commented on config/database.js:
```diff
@@ -35,43 +36,64 @@ module.exports = {
-  |
-  | npm i --save pg
-  |
-  */
```
Why?

RomainLanz commented on bootstrap/app.js:
```diff
@@ -87,26 +87,26 @@ const aliases = {
   'Adonis/Commands/Auth:Setup',
-  'Adonis/Commands/Repl',
+  'Adonis/Commands/DB:Seed',
+  'App/Commands/Greet',
```
This is an example command, it should stick to the top to be removed easily.

ctf0 replied:
will keep it along with greet at the top then

ctf0 committed:
1- changed back the default db
2- removed the extra env db keys
3- added another key for session cookie name as another security measure

ctf0 committed:
move the example cmnds to the top

RomainLanz commented:
I'll let @thetutlage review it and react to changes.

ctf0 commented:
@RomainLanz btw,if i've a couple of ideas what is the best place to put ? i dont think issuing a ticket on the main repo is the correct action

RomainLanz commented:
You should use the https://github.com/adonisjs/discussion repository.

thetutlage reviewed (commented):
@ctf0 Their are some reviews, please look into them so that I can merge the PR

ctf0 committed:
change post data limit

ctf0 committed:
minor edit for auth config

thetutlage commented on .env.example:
```diff
@@ -4,8 +4,6 @@ APP_KEY=n96M1TPG821EdN4mMIjnGKxGytx9W2UJ
 CACHE_VIEWS=false
 SESSION_DRIVER=cookie
+SESSION_DOMAIN=example.com
+SESSION_COOKIE=example-session
```
Session cookie does not needs to be here. Neither the session domain. 

I will never change the `cookie` name for each environment. Also their should be `DOMAIN` variable, which can be used by application internally and should not be part of core

ctf0 replied:
the reason behind `SESSION_COOKIE & SESSION_DOMAIN` is to minimize being effected by any bug that could be used by an attacker which would effect all the apps made by adonis

ctf0 committed:
add services.js

ctf0 committed:
add csp default Starter Policy directives

ctf0 committed:
add Encryption sp and alias

ctf0 committed:
add mail config file

thetutlage commented:
This PR has lot of stuff going. I would suggest to split into multiple PR's
